SimG4CMS/Forward/src/ZdcShowerLibrary.cc

```diff
@@ -203,15 +203,14 @@ int ZdcShowerLibrary::getEnergyFromLibrary(const G4ThreeVector& hitPoint,
     edis = 3.0;
   }
 
-  if (eav < 0. || edis < 0.) {
-    LogDebug("ZdcShower") << " Negative everage energy from parametrization \n"
+  if (eav < 0. || esig < 0.) {
+    LogDebug("ZdcShower") << " Negative everage energy or esigma from parametrization \n"
                           << " xin: " << xin << "(cm)"
                           << " yin: " << yin << "(cm)"
                           << " track en: " << energy << "(GeV)"
-                          << " eaverage: " << eav / GeV << " (GeV)"
-                          << " esigma: " << esig / GeV << "  (GeV)"
-                          << " edist: " << edis << " (GeV)"
-                          << " dE hit: " << nphotons / GeV << " (GeV)";
+                          << " eaverage: " << eav << " (GeV)"
+                          << " esigma: " << esig << "  (GeV)"
+                          << " edist: " << edis << " (GeV)";
```

Comment:
Note that the old debug message was incorrectly trying to convert the `eaverage` and `esigma` values to GeV. They are already in GeV at this point in the code. This debug message had originally been copy and pasted from a few lines down in this function, after the point where `eaverage` and `esigma` had been converted to MeV, which explains why the original message was trying to convert them to GeV.
Also, the "dE hit" line has been deleted from the debug message because at this point in the code `nphotons` still has its initialized value of -1, so there's no point in printing it out.

Reply:
It seems, that "double edis" is not a good solution - it may be Boolean variable.